### PR TITLE
Use picture and srcsets

### LINKED ISSUE #5739: Use <picture> and srcset's for PNI Homepage images
Related issue: #5677

On the PNI Homepage there are product images of each product, but cloudinary is using a 600px width image and non-cloudinary is using regular upload sizes. 

We can make this page better for users by leveraging HTML's `<picture>` element. Something like this:

```html
<picture>
 <source
   media="(min-width: 900px)"
   srcset="large-image_1x.jpeg 1x, large-image_retina.jpeg 2x"
   type="image/jpeg" >
 <source
   media="(min-width: 601px)"
   srcset="medium-image_1x.webp 1x, medium-image_retina.jpeg 2x"
   type="image/jpeg" >
 <source
   media="(max-width: 600px)"
   srcset="small-image_1x.webp 1x, small-image_1x.jpeg 1x"
   type="image/jpeg" >
 <img
   src="large-image_1x.jpg"
   type="image/jpeg"
   alt="my image description">
</picture>
```

If this ticket is addressed _after_ PNI has been moved to Wagtail, we can also output .webp images for Chrome users. The idea here is to serve the best media format and size for each device and device size so we don't overspend the users bandwidth (plus it's a good practice).

### PR DESCRIPTION
**Pull request by KalobTaulien**
Closes #5739

**Link to sample test page**: https://foundation-s-5739-use-p-1j6pev.herokuapp.com/en/privacynotincluded/ (Just the first product is a sample, the rest dont show up due to new PNI stuff, not because of this repo - but feel free to 2x check that!)

### CODE REVIEW

File: network-api/networkapi/buyersguide/templates/buyersguide/catalog.html [hunk 1]
```diff
@@ -80,6 +80,47 @@ <h1 class="h1-heading text-center mt-0">{{ page.header }}</h1>
...
+              <source
+                media="(max-width: 768px)"
+                {% if USE_CLOUDINARY %}
+                  srcset="{% cloudinary_url product.cloudinary_image quality=50 fetch_format="auto" crop="fit" width=358 %} 1x, {% cloudinary_url product.cloudinary_image quality=50 fetch_format="auto" crop="fit" width=716 %} 2x"
+                {% else %}
+                  {% image product.image fill-358x358 as img_1x %}
+                  {% image product.image fill-716x716 as img_2x %}
+                  srcset="{{ img_1x.url }} 1x, {{ img_2x.url}} 2x"
+                {% endif %}
+              >
```
Comment (mmmavis) [inline network-api/networkapi/buyersguide/templates/buyersguide/catalog.html:123]:
```suggestion
              <source
                media="(min-width: 992px)"
                {% if USE_CLOUDINARY %}
                  srcset="{% cloudinary_url product.cloudinary_image quality=50 fetch_format="auto" crop="fit" width=260 %} 1x, {% cloudinary_url product.cloudinary_image quality=50 fetch_format="auto" crop="fit" width=520 %} 2x"
                {% else %}
                  {% image product.image fill-260x260 as img_1x %}
                  {% image product.image fill-520x520 as img_2x %}
                  srcset="{{ img_1x.url }} 1x, {{ img_2x.url}} 2x"
                {% endif %}
              >
              <source
                media="(min-width: 768px)"
                {% if USE_CLOUDINARY %}
                  srcset="{% cloudinary_url product.cloudinary_image quality=50 fetch_format="auto" crop="fit" width=305 %} 1x, {% cloudinary_url product.cloudinary_image quality=50 fetch_format="auto" crop="fit" width=610 %} 2x"
                {% else %}
                  {% image product.image fill-305x305 as img_1x %}
                  {% image product.image fill-610x610 as img_2x %}
                  srcset="{{ img_1x.url }} 1x, {{ img_2x.url}} 2x"
                {% endif %}
              >
              <source
                media="(min-width: 415px)"
                {% if USE_CLOUDINARY %}
                  srcset="{% cloudinary_url product.cloudinary_image quality=50 fetch_format="auto" crop="fit" width=360 %} 1x, {% cloudinary_url product.cloudinary_image quality=50 fetch_format="auto" crop="fit" width=720 %} 2x"
                {% else %}
                  {% image product.image fill-360x360 as img_1x %}
                  {% image product.image fill-720x720 as img_2x %}
                  srcset="{{ img_1x.url }} 1x, {{ img_2x.url}} 2x"
                {% endif %}
              >
              <source
                media="(max-width: 414px)"
                {% if USE_CLOUDINARY %}
                  srcset="{% cloudinary_url product.cloudinary_image quality=50 fetch_format="auto" crop="fit" width=185 %} 1x, {% cloudinary_url product.cloudinary_image quality=50 fetch_format="auto" crop="fit" width=370 %} 2x"
                {% else %}
                  {% image product.image fill-185x185 as img_1x %}
                  {% image product.image fill-370x370 as img_2x %}
                  srcset="{{ img_1x.url }} 1x, {{ img_2x.url}} 2x"
                {% endif %}
              >
```

Comment (mmmavis) [reply network-api/networkapi/buyersguide/templates/buyersguide/catalog.html:123]: Code looks good. I suggest adjusting some pixel values a bit. 

Using Bootstrap's breakpoints as reference:

- 1st set: for browser window > 992px wide, each product grid is roughly 260px wide. This means we don't need to have another set for > 1200px.
- 2nd set: for browser window > 768px wide -- I set my browser window to 991px and measured product grids, each was roughly 305px.
- 3rd set: (an in-between set that I added)
- 4th set: for browser window > 415px wide --- 415px isn't a Bootstrap breakpoint but it's how wide browser window on iPhone Plus is. I'm thinking having this set for mobile sized browser windows can save users some bandwidth.

**Review by mmmavis — CHANGES_REQUESTED** [review]
Added some thoughts.

File: network-api/networkapi/buyersguide/templates/buyersguide/catalog.html (hunk 1 above)
Comment (KalobTaulien) [reply network-api/networkapi/buyersguide/templates/buyersguide/catalog.html:123]: Awesome suggestions! Thank you for adjusting those!

**Review by mmmavis — APPROVED** [review]
🎉

**Pull request merged**